### PR TITLE
Prune workflows based on changed files

### PR DESCRIPTION
**Pull request by KyleFromNVIDIA**
## Description
Contributes to https://github.com/rapidsai/build-planning/issues/94

## Checklist
- [ ] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/rmm/blob/HEAD/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

File: .github/workflows/pr.yaml [hunk 1]
```diff
@@ -24,6 +25,31 @@ jobs:
...
+          # TODO: Remove this before merging
+          - '!.github/workflows/**'
+        test_python:
+          - '**'
+          - '!.devcontainer/**'
+          - '!.pre-commit-config.yaml'
+          - '!CONTRIBUTING.md'
+          - '!README.md'
+          # TODO: Remove this before merging
+          - '!.github/workflows/**'
```
Comment (jameslamb) [inline .github/workflows/pr.yaml:52]:
```suggestion
          - '!README.md'
          - '!img/**'
          - '!python/**'
          # TODO: Remove this before merging
          - '!.github/workflows/**'
        test_python:
          - '**'
          - '!.devcontainer/**'
          - '!.pre-commit-config.yaml'
          - '!CONTRIBUTING.md'
          - '!README.md'
          - '!img/**'
          # TODO: Remove this before merging
          - '!.github/workflows/**'
```

The `img/` folder seems to just contain static images for the docs, so I think it's safe to assume that changes only touching that folder don't require re-runs of the Python or C++ tests.

**Review by jameslamb — APPROVED** [review]
Just one small suggestion, otherwise looks good. I don't need to re-review.

File: .github/workflows/pr.yaml (hunk 1 above)
Comment (KyleFromNVIDIA) [reply .github/workflows/pr.yaml:52]: Ahh, I don't know how I missed that this one had `img/`. I will add it.

**Pull request merged**